lib/couch_potato/persistence/properties.rb

```diff
@@ -11,30 +11,42 @@ class PropertyList
 
         def initialize(clazz)
           @clazz = clazz
-          @list = []
+          @list = {}
         end
 
         def each
-          (list + inherited_properties).each {|property| yield property}
+          (@all_list ||= inherited_properties.merge(list)).values.each {|property| yield property}
         end
 
         def <<(property)
-          @list << property
+          list[property.name] = property
+        end
+        
+        def []=(key, property)
+          list[key] = property
         end
 
+        def [](key)
+          (@all_list ||= inherited_properties.merge(list))[key]
```

Comment:
duplicate code with line 18. also not sure if caching this could lead to problems. in general the performance benefit should be minimal anyway since properties are only initialized once on app startup.